### ISSUE TITLE
Ho can i get functions calls?

### ISSUE THREAD

**Issue by hwcibreiro** [SciSharp/LLamaSharp#869]

**Comment by sangyuxiaowu** [comment]
This should require upper level applications to implement.

**Comment by martindevans** [comment]
llama.cpp and LLamaSharp don't currently have any function calling support. That will probably change in the future :)

**Comment by sangyuxiaowu** [comment]
The [LLamaWorker](https://github.com/sangyuxiaowu/LLamaWorker) project based on the LLamaSharp, has preliminarily implemented function calling.

**Comment by martindevans** [comment]
@sangyuxiaowu that looks really cool - if you'd like to submit a PR adding a link to it in [this section of the readme](https://github.com/SciSharp/LLamaSharp?tab=readme-ov-file#integrations--examples) that would be great :D

**Comment by hwcibreiro** [comment]
Thank you

**Comment by sangyuxiaowu** [comment]
> @sangyuxiaowu that looks really cool - if you'd like to submit a PR adding a link to it in [this section of the readme](https://github.com/SciSharp/LLamaSharp?tab=readme-ov-file#integrations--examples) that would be great :D

I have already added it to the relevant project application examples before, does it seem that more introduction is needed?

Currently, it is:  [LLamaWorker (ASP.NET Web API like OAI)](https://github.com/sangyuxiaowu/LLamaWorker)
Maybe:  LLamaWorker (ASP.NET Web API like OAI and Function Calling Support) ？

I tested the performance of Llama3.1 today and it seems to be good. Multiple tools can be called in parallel.

![image](https://github.com/user-attachments/assets/0b2bd381-54b0-4835-8af3-cf77cc662ed1)
